### DISCUSSION TITLE
How is the loss evaluated?

### DISCUSSION THREAD

**Discussion by edoardo100** [lululxvi/deepxde#1033] [Q&A]
I have a question related to the loss evaluation in deepXDE.

Let us take the example for deepONet in https://deepxde.readthedocs.io/en/latest/demos/operator/antiderivative_aligned.html.
If I understand properly the network is defined in
```

m = 100
dim_x = 1
net = dde.nn.DeepONetCartesianProd(
    [m, 40, 40],
    [dim_x, 40, 40],
    "relu",
    "Glorot normal",
)
```

and the loss will have a term like `y_true-y_pred`, with `y_true` and `y_pred` arrays representing the pde solution as calculated a priori and the predicted solution. Now, if my interpretation is correct, `y_true` should have length equal to 100 along one axis, while `y_pred`, in the example provided, will be of length 40 along the same axis. 
My question is: how is the loss actually evaluated? Is there a sort of interpolation step for one of the vectors involved or am I missing something?

Thank you in advance

**Comment by lululxvi** [comment]
See the papers: https://doi.org/10.1038/s42256-021-00302-5, https://doi.org/10.1016/j.cma.2022.114778
